src/rust/ide/json-rpc/src/macros.rs

```diff
@@ -0,0 +1,158 @@
+//! Helper macros to generate RemoteClient and MockClient.
+
+
+
+// ====================
+// === Helper macro ===
+// ====================
+
+/// Generates parameters maps. e.g.:
+/// `(age:u32)` => `u32`
+/// `(age:u32,height:f32)` => `(u32,f32)`
+#[macro_export]
+macro_rules! make_param_map {
+    (,$ty:ty) => {
+        $ty
+    };
+    (,$($ty:ty),+) => {
+        ($($ty),+)
+    }
+}
```

Comment:
In no waythis macro does what the docs tell it does. This is very bad code style! Moreover this macro is a big hack. It should be removed and the macro which uses it should be fixed instead.